source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md

```diff
@@ -0,0 +1,74 @@
+
+  * It ensures a certain level of quality on the master branch, since we have to ensure that CI jobs are passing.
+  * In encourages peer reviews, so code should always be checked by more than one person before merging.
+  
+- **Use the cloud**: it would not make much sense to develop a cloud operating system if we could not use the cloud ourselves, would it? As expected, all the OpenStack infrastructure is hosted in [OpenStack-based clouds donated by different companies](https://www.openstack.org/foundation/companies/#infra-donors). Since the infrastructure deployment and configuration is automated, it is quite easy to manage in a cloud environment. And as we will see later, it is also a perfect match for our continuous integration processes.
+
+- **Automated continuous integration**: this is a key part of the development process in the OpenStack community. Each project defines a number of CI jobs, covering unit and integration tests. With 5000 to 8000 commits a month, it would be impossible to cover all those tests without solid automation. So our process includes tooling that will spawn short-lived VMs in one of the OpenStack-based clouds, run the defined tests, and delete the VM when done. This, together with the requirement for CI jobs to pass before merging any code, minimizes the amount of regressions in our codebase.
+  
+- **Use (and contribute to) Open Source**: one of the "Four Opens" that drive the OpenStack community is [Open Source](https://docs.openstack.org/project-team-guide/open-source.html). As such, all of the development and infrastructure processes happen using Open Source software. And not just that, the OpenStack community has created several libraries and applications with great potential for reuse outside the OpenStack use case. Applications like [Zuul](https://docs.openstack.org/infra/zuul/) and [nodepool](https://docs.openstack.org/infra/nodepool/index.html), general-purpose libraries like [pbr](https://docs.openstack.org/developer/pbr/), or the contributions to the [SQLAlchemy](https://www.sqlalchemy.org/) library are good examples of this.
+
+## Tools
+
+So, which tools do we use to make all of this happen? As stated above, the OpenStack community relies on several open source tools to do its work:
+    
+- **Infrastructure as code**
+  * Git to store the infrastructure definitions
+  * Puppet and Ansible as configuration management and orchestration tools
+- **Development**
+  * Git as a code repository
+  * Gerrit as a code review and repository management tool
+  * Etherpad as a collaborative editing tool
+- **Continuous integration**
+  * Zuul as an orchestrator of the gate checks
+  * Nodepool to automate the creation and deletion of short-lived VMs for CI jobs across multiple clouds
+  * Jenkins to execute CI jobs (actually, it has now been replaced by Zuul itself)
+
```

Comment:
Tempest is also used as an integration and validation suite. Do we need to mention about it?